delocate/fuse.py

```diff
@@ -39,6 +44,42 @@ def _copyfile(in_fname, out_fname):
     os.chmod(out_fname, perms)
 
 
+def _retag_wheel(to_wheel: Path, from_wheel: Path, to_tree: Path) -> str:
+    """Update the name and dist-info to reflect a univeral2 wheel.
+
+    Parameters
+    ----------
+    to_wheel : Path
+        The path of the wheel to fuse into.
+    from_wheel : Path
+        The path of the wheel to fuse from.
+    to_tree : Path
+        The path of the directory tree to fuse into (update into).
+
+    Returns
+    -------
+    retag_name : str
+        The new, retagged name the out wheel should be.
+    """
+    to_tree = to_tree.resolve()
```

Comment:
Does `to_tree` need to be resolved here? It seems it's mainly passed to other functions from here which should handle validation on their own.

Reply:
I'm not sure? Strictly speaking I guess no, a relative path here would work fine, thats what it was originally. But since `to_wheel` and `from_wheel` were resolved I thought it should be resolved at some point. Also, I don't think the other functions that use this do resolve it themselves (i.e. `_check_and_update_wheel_name` and `_update_wheelfile`).

Something else I had considered was using path instead of str in `fuse_wheels` for the trees (`"to_wheel", "from_wheel"`. Then I could just resolve them on creation and pass them in to all the functions in the `InTemporaryDirectory` context. This way it we would be passing the object around rather than rewriting the string each time. But I didn't do it since some of those functions don't accept os.PathLike. I could update those functions as well though (or convert the Path object to a string and then pass them in...but I don't love that idea).

Reply:
Without `strict=True` there'd be no real reason to resolve `to_wheel`/`from_wheel` either, and even that can be handled by the functions farther down. Normalizing the paths is only needed for a few specific things, and any functions which require their paths normalized should do it themselves rather than rely on the caller.

Some confusion might come from `InTemporaryDirectory` abusing the working directory. I'd personally rather have the relative paths be more strongly associated with the temporary directory. Or at least I don't immediately realize `"to_wheel"` is supposed to be a realtive path whenever I see it.
```py
with tempfile.TemporaryDirectory() as temp_dir:
    to_wheel_dir = Path(temp_dir, "to_wheel")
    from_wheel_dir = Path(temp_dir, "from_wheel")
```

Feel free to add `os.PathLike` support to any function needed. If the body is too complex to refactor then you could always do `str(Path(param))`.

Reply:
Thanks for explaining all that. I sometimes, just in general not only in delocate, am unsure if I should be using relative or resolved paths, when either will work. But I think the strategy, only when it is needed, makes sense.

> I'd personally rather have the relative paths be more strongly associated with the temporary directory.

So if I'm understanding what you said correctly, you are okay with using `tempfile.TemporaryDirectory()` rather than something from `delocate.tempdirs`?

Reply:
> So if I'm understanding what you said correctly, you are okay with using `tempfile.TemporaryDirectory()` rather than something from `delocate.tempdirs`?

Yes.

Decloate used to be a Python 2 package, so it's full of backports of features that are now native to Python 3. This means the stuff in `delocate.tempdirs` is outdated.